dateutil/parser/isoparser.py

```diff
@@ -0,0 +1,380 @@
+# -*- coding: utf-8 -*-
+"""
+This module offers a parser for ISO-8601 strings
+
+It is intended to support all valid date, time and datetime formats per the
+ISO-8601 specification.
+"""
+from datetime import datetime, timedelta, time, date
+import calendar
+from dateutil import tz
+
+from functools import wraps
+
+import re
+import six
+
+__all__ = ["isoparse", "isoparser"]
+
+
+def _takes_ascii(f):
+    @wraps(f)
+    def func(self, str_in, *args, **kwargs):
+        # If it's a stream, read the whole thing
+        str_in = getattr(str_in, 'read', lambda: str_in)()
+
+        # If it's unicode, turn it into bytes, since ISO-8601 only covers ASCII
+        if isinstance(str_in, six.text_type):
+            # ASCII is the same in UTF-8
+            try:
+                str_in = str_in.encode('ascii')
+            except UnicodeEncodeError as e:
+                msg = 'ISO-8601 strings contain only ASCII characters'
```

Comment:
I might say something like "should contain" from the error it might not be clear that the issue is the input string containing "invalid" characters

Reply:
OK, saw the raise from, still would change "contain" to should contain if string refers to the input string

Reply:
It refers to ISO-8601 strings. All ISO-8601 strings consist of only ASCII characters, even though there are invalid things *intending* to be ISO-8601 strings that contain non-ascii characters.